lib/src/instances/option.dart

```diff
@@ -2,52 +2,172 @@ import 'package:shuttlecock/shuttlecock.dart';
+  }
+
+  @override
+  int get length => 0;
+
+  @override
+  T get single {
+    throw new StateError("No element");
+  }
+
+  @override
   Option<T> operator +(Option<T> other) => other;
 
   @override
   bool operator ==(Object other) => identical(this, other) || other is None;
 
   @override
-  Option<S> app<S>(Option<Function1<T, S>> app) => new None();
+  bool any(bool f(T element)) => false;
+
+  @override
+  Option<S> app<S>(Option<Function1<T, S>> app) => none;
+
+  @override
+  bool contains(Object element) => false;
+
+  @override
+  T elementAt(int index) {
+    throw new RangeError.index(index, this, "index", null, 0);
+  }
+
+  @override
+  bool every(bool f(T element)) => true;
+
+  @override
+  IterableMonad<S> expand<S>(Iterable<S> f(T element)) =>
+      new IterableMonad.fromIterable(<S>[]);
+
+  @override
+  T firstWhere(bool test(T element), {T orElse()}) {
+    if (orElse == null) {
+      throw new StateError('Empty collection.');
+    }
+
+    return orElse();
+  }
 
   @override
   Option<S> flatMap<S>(Function1<T, Monad<S>> f) => new None();
 
   @override
+  S fold<S>(S initialValue, S combine(S previousValue, T element)) =>
+      initialValue;
+
+  @override
+  void forEach(void f(T element)) {}
+
+  @override
+  String join([String separator = ""]) => '';
+
+  @override
+  T lastWhere(bool test(T element), {T orElse()}) {
+    if (orElse == null) {
+      throw new StateError('Empty collection.');
+    }
+
+    return orElse();
+  }
+
+  @override
   Option<S> map<S>(Function1<T, S> f) => new None<S>();
 
   @override
+  T reduce(T combine(T value, T element)) {
+    throw new StateError("Too few elements");
+  }
+
+  // TODO: implement single
+  @override
+  T singleWhere(bool test(T element)) {
+    throw new StateError("No element");
+  }
+
+  @override
+  IterableMonad<T> skip(int count) => new IterableMonad.fromIterable([]);
+
+  @override
+  IterableMonad<T> skipWhile(bool test(T value)) =>
+      new IterableMonad.fromIterable([]);
+
+  @override
+  IterableMonad<T> take(int count) {
+    throw new StateError("Too few elements");
+  }
+
+  @override
+  IterableMonad<T> takeWhile(bool test(T value)) => this;
+
+  @override
+  List<T> toList({bool growable: true}) => [];
+
+  @override
+  Set<T> toSet() => new Set.identity();
+
+  @override
   String toString() => 'None';
+
+  @override
+  IterableMonad<T> where(bool test(T element)) => this;
 }
 
 /// Represents an optional value. It satisfies the type equation FX = 1 + X,
 /// where the functor F takes a set to a point plus that set. This is known as
 /// Maybe in Haskell.
-abstract class Option<T> extends Monad<T> implements Monoid<T> {
+abstract class Option<T> extends Monad<T>
+    implements Monoid<T>, IterableMonad<T> {
   /// Creates a new option provided its value, of the value is null, it will
   /// return None.
   factory Option(T value) {
     if (value == null) {
-      return new None();
+      return None.none;
```

Comment:
now that `isEmpty` is available, I think it will be more explicit to use it instead of checking if the value is null